lib/cache.ts

```diff
@@ -0,0 +1,115 @@
+import fs from 'fs';
+import os from 'os';
+import { DEFAULT_NETWORK, SupportedNetworks } from 'modules/web3/constants/networks';
+import { config } from 'lib/config';
+import Redis from 'ioredis';
+import packageJSON from '../package.json';
+
+const redis = config.REDIS_URL
+  ? new Redis(config.REDIS_URL, {
+      connectTimeout: 10000
+    })
+  : null;
+
+const oneHourInMS = 60 * 60 * 1000;
+
+// Mem cache does not work on local instances of nextjs because nextjs creates clean memory states each time.
+const memoryCache = {};
+
+function getFilePath(name: string, network: string): string {
+  const date = new Date().toISOString().substring(0, 10);
+
+  return `${os.tmpdir()}/gov-portal-version-${packageJSON.version}-${network}-${name}-${date}`;
+}
+
+export const cacheDel = (path: string): void => {
+  console.log('Delete cache', path);
+  fs.unlinkSync(path);
+  memoryCache[path] = null;
+};
+
+export const cacheGet = async (
+  name: string,
+  network?: SupportedNetworks,
+  expiryMs?: number
+): Promise<any> => {
+  const isRedisCache = !!config.REDIS_URL;
+
+  try {
+    const currentNetwork = network || DEFAULT_NETWORK.network;
+    const path = getFilePath(name, currentNetwork);
+
+    if (isRedisCache && redis) {
+      // Get redis data if it exists
+      const cachedData = await redis.get(path);
```

Comment:
What happens here if `path` doesn't exist? Does it return null or throw an error?

Reply:
yes, returns null